File: 2022/python/day6.py
def run(line, step):
	return [i for i in range(step, len(line), 1) if len(set(line[i-step:i])) == step][0]

with open('../inputs/input6') as f:
	lines = [line.strip() for line in f.readlines() if len(line) > 0]
	assert(len(lines) > 0)
	print('Part 1:', run(lines[0], 4))
	print('Part 2:', run(lines[0], 14))

File: 2022/ts/day6.ts
import { assert } from "./deps.ts";
import { readLines } from "./utils.ts";

function run(step: number): number {
  const lines = readLines("../inputs/input6").filter((line) => line.length > 0);
  assert(lines.length > 0);
  const line = lines[0];

  for (let i = step; i < line.length; ++i) {
    const uniqueCharacters = new Set(line.substring(i - step, i).split(""));
    if (uniqueCharacters.size === step) {
      return i;
    }
  }
  return 0;
}

function part1() {
  console.log(`Part 1: ${run(4)}`);
}

part1();

function part2() {
  console.log(`Part 2: ${run(14)}`);
}

part2();
